File: my-first-app/app/src/main/java/org/feup/ddmm/myfirstapp/SecondActivity.java
package org.feup.ddmm.myfirstapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;
import android.content.Intent;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Intent intent = getIntent();
        String message = intent.getStringExtra(FirstActivity.EXTRA_MESSAGE);

        TextView textView = new TextView(this);
        textView.setTextSize(30f);
        textView.setText(message);

        setContentView(textView);
    }
}

File: MyFirstApp/app/src/main/java/org/feup/ddmm/myfirstapp/SecondActivity.kt
package org.feup.ddmm.myfirstapp

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView

class SecondActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        var message = intent.getStringExtra(FirstActivity.EXTRA_MESSAGE)
        var textView = TextView(this)
        textView.textSize = 30f
        textView.text = message

        setContentView(textView)
    }
}
